### PR TITLE
MT940 to ISO transformer and MT940 to JSON transformer

### PR DESCRIPTION
**Pull request by Mohamedizzath**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 1]
```diff
@@ -138,6 +138,11 @@ private Map<String, Object> extractFields(String[] lines) {
             if (line.startsWith(OPEN_BRACKET) || line.startsWith(DASH) || line.length() < 3) {
                 continue;
             }
+
+            if (line.endsWith("&#xd;")) {
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:142]: Shall we add a comment here mentioning why we are doing this

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            textBlock.getAsJsonObject(balanceType).remove("amount");
+            textBlock.getAsJsonObject(balanceType).addProperty("amount", balanceAmt);
+        }
+    }
+
+    /**
+     * Update the dates in the MT940 for processing freemarker template to MT940 to ISO convert.
+     * @param mt940JsonObj         JSON object tree for MT940 message
+     */
+    public static void updateDatesFrMT940(JsonObject mt940JsonObj) {
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java:92]: Check for spellings

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        if (mt940JsonObj.has("applicationHeaderBlock")) {
+            String inputTime = mt940JsonObj.getAsJsonObject("applicationHeaderBlock")
+                    .get("inputTime").getAsString();
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java:95]: Shall we move these strings as constants

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java (hunk 1 above)
Comment (Mohamedizzath) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:143]: Added in e2eba5896d2549c41ebd26848c9e0af9148a62ce

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java (hunk 2 above)
Comment (Mohamedizzath) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java]: Fixed in e2eba5896d2549c41ebd26848c9e0af9148a62ce

File: src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java (hunk 3 above)
Comment (Mohamedizzath) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940JSONParserUtils.java]: Updated in e2eba5896d2549c41ebd26848c9e0af9148a62ce

**Review by Ashi1993 — APPROVED** [review]

**Review by lalaji — APPROVED** [review]

**Pull request merged**